### ISSUE TITLE
Return type of Byte[] fail

### ISSUE THREAD

**Issue by sonicmouse** [jacqueskang/IpcServiceFramework#17]
I haven't looked in to why .. will update when i figure it out. Just a heads up in case i get side-tracked with other work.

**Comment by sonicmouse** [comment]
OK i looked in to it. When you serialize a byte array with `TypeNameHandling = TypeNameHandling.Objects` on, it converts it to a `JObject` that resembles this:
```
{
    "$value":"<<base64encodedData>>"
}
```
so, because of the `$value`, `JObject::ToObject()` fails. No idea how to fix this properly, so this may be a kludge:
```
            if (origValue is JObject jObj)
            {
                if (jObj.TryGetValue("$value", out JToken jTok))
                {
                    destValue = JsonConvert.DeserializeObject(JsonConvert.SerializeObject(jTok), destType);
                    return true;
                }
                destValue = jObj.ToObject(destType);
                return true;
            }
```
It fixes it on my end... but there may be a better way. Perhaps there is a way for the deserializer to be aware the `$value` parameter exists?

**Comment by jacqueskang** [comment]
Resolved in v1.0.4.3

**Comment by heartacker** [comment]
hi, I have the same issue when return byte[]

```
  <package id="JKang.IpcServiceFramework.Client.NamedPipe" version="3.1.0" targetFramework="net472" />
```
![image](https://user-images.githubusercontent.com/1876302/163982253-a6c3b0fd-abec-4be5-a126-c13b4da48e0a.png)

can anybody help me to fix this
